### PR TITLE
Clean up the RDM Discovery Agent.

### PR DESCRIPTION
**Pull request by nomis52**
- Send a broadcast unmute 3 times in case a device misses it.
- Support trailing data for DUB responses.
- Clean up the documentation.

### CODE REVIEW

File: common/rdm/DiscoveryAgent.cpp [hunk 1]
```diff
@@ -250,25 +210,35 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
...
     return;
   }
 
-  unsigned int preamble_size = length - MIN_DUB_RESPONSE_SIZE;
-  for (unsigned int i = 0; i < preamble_size; i++) {
-    if (data[i] != 0xfe) {
-      OLA_INFO << "preamble " << i << " " << std::hex <<
-        static_cast<int>(data[i]);
+  unsigned int offset = 0;
+  while (data[offset] != 0xaa && offset < PREAMBLE_SIZE - 1) {
```
Comment (peternewman) [inline common/rdm/DiscoveryAgent.cpp]: Can we have 0xaa as a constant?

File: common/rdm/DiscoveryAgent.cpp [hunk 2]
```diff
@@ -250,25 +210,35 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
...
+    if (data[offset] != 0xfe) {
```
Comment (peternewman) [inline common/rdm/DiscoveryAgent.cpp]: Likewise 0xfe.

File: common/rdm/DiscoveryAgent.cpp [hunk 3]
```diff
@@ -250,25 +210,35 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
...
+      OLA_INFO << "preamble " << offset << " " << strings::ToHex(data[offset]);
       HandleCollision();
       return;
     }
+    offset++;
+  }
+
+  if (data[offset] != 0xaa) {
```
Comment (peternewman) [inline common/rdm/DiscoveryAgent.cpp]: Again

File: common/rdm/DiscoveryAgent.cpp [hunk 4]
```diff
@@ -250,25 +210,35 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
...
+    return;
   }
 
-  unsigned int offset = preamble_size;
-  if (data[offset++] != 0xaa) {
-    OLA_INFO << "preamble separator is " << std::hex <<
-      static_cast<int>(data[offset]);
+  offset++;
+  unsigned int remaining = length - offset;
+  OLA_INFO << "length was " << length << ", offset " << offset
```
Comment (peternewman) [inline common/rdm/DiscoveryAgent.cpp]: SPaG Length

File: common/rdm/DiscoveryAgent.cpp [hunk 5]
```diff
@@ -250,25 +210,35 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
...
 
-  unsigned int preamble_size = length - MIN_DUB_RESPONSE_SIZE;
-  for (unsigned int i = 0; i < preamble_size; i++) {
-    if (data[i] != 0xfe) {
-      OLA_INFO << "preamble " << i << " " << std::hex <<
-        static_cast<int>(data[i]);
+  unsigned int offset = 0;
+  while (data[offset] != 0xaa && offset < PREAMBLE_SIZE - 1) {
+    if (data[offset] != 0xfe) {
+      OLA_INFO << "preamble " << offset << " " << strings::ToHex(data[offset]);
```
Comment (peternewman) [inline common/rdm/DiscoveryAgent.cpp]: SPaG Preamble

File: include/ola/rdm/DiscoveryAgent.h [hunk 6]
```diff
@@ -32,129 +32,222 @@
...
-    static const unsigned int MAX_EMPTY_BRANCH_ATTEMPTS = 5;
-    /*
-     * The maximum number of times we'll perform discovery on a branch when we
-     * get an inconsistent result (responder not muting, etc.)
-     */
-    static const unsigned int MAX_BRANCH_FAILURES = 5;
-    // The number of times we'll attempt to mute a UID
-    static const unsigned int MAX_MUTE_ATTEMPTS = 5;
+  /**
+   * Represents a range of UIDs (a branch of the UID tree)
```
Comment (peternewman) [inline include/ola/rdm/DiscoveryAgent.h]: @brief

File: common/rdm/DiscoveryAgent.cpp (hunk 1 above)
Comment (nomis52) [inline common/rdm/DiscoveryAgent.cpp]: Done

File: common/rdm/DiscoveryAgent.cpp (hunk 2 above)
Comment (nomis52) [inline common/rdm/DiscoveryAgent.cpp]: Done

File: common/rdm/DiscoveryAgent.cpp (hunk 5 above)
Comment (nomis52) [inline common/rdm/DiscoveryAgent.cpp]: Done

File: common/rdm/DiscoveryAgent.cpp (hunk 3 above)
Comment (nomis52) [inline common/rdm/DiscoveryAgent.cpp]: Done

File: common/rdm/DiscoveryAgent.cpp (hunk 4 above)
Comment (nomis52) [inline common/rdm/DiscoveryAgent.cpp]: Removed

File: include/ola/rdm/DiscoveryAgent.h (hunk 6 above)
Comment (nomis52) [inline include/ola/rdm/DiscoveryAgent.h]: Done

**Pull request merged**